test/ng/ngPropSpec.js

```diff
@@ -0,0 +1,775 @@
+          $rootScope.testUrl = $sce.trustAsMediaUrl('javascript:foo()');
+          $rootScope.$digest();
+          expect(element.find(tag).prop('src')).toEqual('javascript:foo()');
+
+          // As a URL
+          element = $compile('<video><' + tag + ' ng-prop-src="testUrl"></' + tag + '></video>')($rootScope);
+          $rootScope.testUrl = $sce.trustAsUrl('javascript:foo()');
+          $rootScope.$digest();
+          expect(element.find(tag).prop('src')).toEqual('javascript:foo()');
+
+          // As a RESOURCE URL
+          element = $compile('<video><' + tag + ' ng-prop-src="testUrl"></' + tag + '></video>')($rootScope);
+          $rootScope.testUrl = $sce.trustAsResourceUrl('javascript:foo()');
+          $rootScope.$digest();
+          expect(element.find(tag).prop('src')).toEqual('javascript:foo()');
+        }));
+      });
+    });
+  }
+
+  describe('img[src] sanitization', function() {
+
+    it('should accept trusted values', inject(function($rootScope, $compile, $sce) {
+      var element = $compile('<img ng-prop-src="testUrl"></img>')($rootScope);
+      // Some browsers complain if you try to write `javascript:` into an `img[src]`
+      // So for the test use something different
+      $rootScope.testUrl = $sce.trustAsMediaUrl('someuntrustedthing:foo();');
+      $rootScope.$digest();
+      expect(element.prop('src')).toEqual('someuntrustedthing:foo();');
+    }));
+
+    it('should use $$sanitizeUri', function() {
+      var $$sanitizeUri = jasmine.createSpy('$$sanitizeUri').and.returnValue('someSanitizedUrl');
+      module(function($provide) {
+        $provide.value('$$sanitizeUri', $$sanitizeUri);
+      });
+      inject(function($compile, $rootScope) {
+        var element = $compile('<img ng-prop-src="testUrl"></img>')($rootScope);
+        $rootScope.testUrl = 'someUrl';
+
+        $rootScope.$apply();
+        expect(element.prop('src')).toMatch(/^http:\/\/.*\/someSanitizedUrl$/);
+        expect($$sanitizeUri).toHaveBeenCalledWith($rootScope.testUrl, true);
+      });
+    });
+
+    it('should not use $$sanitizeUri with trusted values', function() {
+      var $$sanitizeUri = jasmine.createSpy('$$sanitizeUri').and.throwError('Should not have been called');
+      module(function($provide) {
+        $provide.value('$$sanitizeUri', $$sanitizeUri);
+      });
+      inject(function($compile, $rootScope, $sce) {
+        var element = $compile('<img ng-prop-src="testUrl"></img>')($rootScope);
+        // Assigning javascript:foo to src makes at least IE9-11 complain, so use another
+        // protocol name.
+        $rootScope.testUrl = $sce.trustAsMediaUrl('untrusted:foo();');
+        $rootScope.$apply();
+        expect(element.prop('src')).toBe('untrusted:foo();');
+      });
+    });
+  });
+
+  ['img', 'source'].forEach(function(srcsetElement) {
+    describe(srcsetElement + '[srcset] sanitization', function() {
```

Comment:
These seem to fail in Edge for `source[srcset]` because the `srcset` property remains `undefined`. Any ideas why assigning this property in Edge would be rejected and remain `undefined`?

Reply:
See https://github.com/angular/angular.js/commit/9de0115711d0963c85eef2faa81242234a5c5f15?w=1